lib/helpers/clone.js

```diff
@@ -41,6 +41,17 @@ function clone(obj, options, isArrayChild) {
     if (options && options._skipSingleNestedGetters && obj.$isSingleNested) {
       options = Object.assign({}, options, { getters: false });
     }
+    if (options != null && options.retainDocuments != null && obj.$__ != null) {
```

Comment:
this condition seems out of style with the other if's around it, any specific reason everything is tested against null?
(minor style suggestion)

Reply:
Good call, this is a bug, `options.retainDocuments != null` is incorrect because `options.retainDocuments === false` should turn off `retainDocuments` functionality.